mm2src/coins/eth.rs

```diff
@@ -2439,7 +2387,7 @@ async fn sign_transaction_with_keypair(
     let _nonce_lock = coin.nonce_lock.lock().await;
     status.status(tags!(), "get_addr_nonce…");
     let (nonce, web3_instances_with_latest_nonce) = try_tx_s!(
-        get_addr_nonce(coin.my_address, coin.web3_instances.clone())
+        get_addr_nonce(coin.my_address, coin.web3_instances.lock().await.to_vec())
```

Comment:
Good catch. Idea of websocket connections and the way `get_addr_nonce` works conflicts. I don't want to start all the socket connections and keep them alive just for fetching nonce status. I need sometime to find a feasible solution on this.

Reply:
Theorically, if we add oneshot (starting connection just for one request-response and then closing it) mechanism to websocket transport, it would solve this problem. This could also fix the need of spawning connection loops at the coins activation. I will implement and test it during the week.

Reply:
In practice, oneshot mechanism didn't work well. Instead, I implemented temporary connections which gets dropped after the given amount of time.